### PR TITLE
[DEV-3279] as2.0 award amount tooltips & overspending row in Award Amounts Table

### PR DESCRIPTION
**Pull request by MaxwellKendall**
### ~DNM until Exceeds Current & Potential Tooltip copy's are defined~

**High level description:**
Adds tooltips to award amounts visualization chart and overspending row to award amounts table.

**Technical details:**
- Creates component `AwardAmountsChart` to reuse logic for rendering chart with tooltips.
- Improves `AwardAmountsTable` to handle overspending
- Updates CoreAward Model to include overspending and extreme overspending properties

**JIRA Ticket:**
[DEV-3279](https://federal-spending-transparency.atlassian.net/browse/DEV-3279)

**Mockup:**
https://bahdigital.invisionapp.com/share/KYIAAD5E57G#/screens

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] Design review (if applicable)
- [x] Verified cross-browser compatibility

### CODE REVIEW

File: src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx [hunk 1]
```diff
@@ -5,83 +5,56 @@
...
-                    <div className="award__heading-id">
-                        <div className="award__heading-label">{this.props.overview.id ? 'ID' : ''}</div>
-                        <div>{this.props.overview.id}</div>
-                    </div>
-                </div>
-                <hr className="award__divider" />
-                <div className="award__row award-overview" id="award-overview">
+            <AwardPageWrapper
+                identifier={awardId}
+                glssaryLink={glossaryLink}
```
Comment (jonhill13) [inline src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx]: glossaryLink

File: src/js/components/sharedComponents/TooltipWrapper.jsx [hunk 2]
```diff
@@ -91,41 +94,54 @@ export default class TooltipWrapper extends React.Component {
...
-                : window.innerWidth - (tooltipContainer.offsetLeft + tooltipContainer.clientWidth) - 100;
+            tooltipWidth = (spaceToRight > 800)
+                ? 700
+                : spaceToRight - 100;
         }
 
-        let offsetRight = window.innerWidth - tooltipContainer.offsetLeft - tooltipContainer.clientWidth - tooltipWidth - this.props.offsetAdjustments.right;
         if (this.props.left) {
-            offsetRight = (window.innerWidth - tooltipContainer.offsetLeft) + tooltipContainer.clientWidth;
+            const startingPositionLeft = spaceToLeft - tooltipWidth; // minus tooltipWidth b/c right corner of toolTip is flush w/ left edge of toolTip container
```
Comment (jonhill13) [inline src/js/components/sharedComponents/TooltipWrapper.jsx]: can we move this comment above this variable. Adding lines greater than the length for ESLint just adds more warnings to scroll through when finding errors in travis.

File: src/js/components/sharedComponents/TooltipWrapper.jsx [hunk 3]
```diff
@@ -91,41 +94,54 @@ export default class TooltipWrapper extends React.Component {
...
+            this.setState({
+                offsetTop,
+                offsetLeft: startingPositionLeft - horizontalPadding,
+                width: tooltipWidth
+            });
+        }
+        else {
+            const startingPositionLeft = spaceToLeft + ttContainerWidth; // plus ttContainerWidth b/c left corner of toolTip is flush w/ right edge of toolTip container
```
Comment (jonhill13) [inline src/js/components/sharedComponents/TooltipWrapper.jsx]: can we move this comment above this variable. Adding lines greater than the length for ESLint just adds more warnings to scroll through when finding errors in travis.

File: src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/components/awardv2/financialAssistance/FinancialAssistanceContent.jsx]: yes indeed... jonhill13

File: src/js/components/awardv2/contract/ContractContent.jsx [hunk 4]
```diff
@@ -40,7 +31,7 @@ export default class ContractContent extends React.Component {
             : null;
         const awardAmountData = Object.create(BaseAwardAmounts);
         awardAmountData.populate(overview, overview.category);
-
+        console.log("data", awardAmountData);
```
Comment (jonhill13) [inline src/js/components/awardv2/contract/ContractContent.jsx]: can we remove this

**Pull request merged**